neo4j-cypher-dsl-examples/neo4j-cypher-dsl-examples-parser/src/test/java/org/neo4j/cypherdsl/examples/parser/StatementCatalogBuildingVisitorViaParserTest.java

```diff
@@ -158,4 +160,13 @@ void retrievalOfRelationshipsAndTargetSourcesShouldWork() {
 		assertThat(catalog.getSourceNodes(StatementCatalog.type("UNDIRECTED")))
 			.isEmpty();
 	}
+
+	@Test // GH-738
+	void literalsOnParsedStatement() {
+
+		var stmt = CypherParser.parse("LOAD CSV FROM 'https://test.com/test.csv' AS x WITH x MERGE (n {x: x}) ON CREATE SET x.y = NULL, x.a = 'Hallo' ON CREATE SET x.b = [true][1..2] RETURN x");
+		assertThat(stmt.getCatalog().getLiterals())
+			.map(Literal::asString)
+			.containsExactlyInAnyOrder("1", "2", "NULL", "true", "'Hallo'");
```

Comment:
@nmervaillie  would that work for you?

Reply:
@michael-simons I tested it with a bunch of statements. Looks good overall. 
One question with function calls though: things like `CALL dbms.routing.getRoutingTable($routingContext, $databaseName)` returns a set of string literals `['dbms', 'routing']`
Is it expected?

Reply:
No.